### PR TITLE
[release-0.42] placement-configuration, Allow day2 changes

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
Currently placement-configuration should not allow
day2 changes.
Changed to allow day2 changes

**Special notes for your reviewer**:

**Release note**:

```release-note
Allow placement-configuration day2 changes
```

### CODE REVIEW

File: pkg/network/placement_configuration_test.go [hunk 1]
```diff
@@ -49,95 +49,34 @@ var _ = Describe("Testing PlacementConfiguration", func() {
 				Infra:     defaultPlacementConfiguration.Infra,
 			},
 		}),
-		Entry("When the user hasn't explicitly changed PlacementConfiguration and a previous configuration exits should use the previous values on the current one", fillDefaultsCase{
-			previousConfig: &cnao.NetworkAddonsConfigSpec{
+		Entry("When Infra PlacementConfiguration is not nil should keep old values", fillDefaultsCase{
```
Comment (rhrazdil) [inline pkg/network/placement_configuration_test.go:52]: The title doesn't describe what is happening imo, we have null PlacementConfiguration and update it with non-null values, and we keep the new values, right?

**Review by rhrazdil — CHANGES_REQUESTED** [review]

File: pkg/network/placement_configuration_test.go (hunk 1 above)
Comment (RamLavi) [reply pkg/network/placement_configuration_test.go:52]: yes you are correct. will fix
Done

File: test/e2e/workflow/deployment_test.go [hunk 2]
```diff
@@ -130,6 +130,57 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			components = append(components, MacvtapComponent)
 			testConfigUpdate(configSpec, components)
 		})
+		Context("and Placement-Configuration is deployed", func() {
```
Comment (rhrazdil) [inline test/e2e/workflow/deployment_test.go:133]: Could you remove the `-` please?

File: test/e2e/workflow/deployment_test.go [hunk 3]
```diff
@@ -130,6 +130,57 @@ var _ = Describe("NetworkAddonsConfig", func() {
...
+
+				By("Checking that deployed components remain up and running")
+				CheckConfigCondition(ConditionAvailable, ConditionTrue, CheckImmediately, time.Minute)
+
+				By("Checking Placement-Configuration was updated")
+				config := GetConfig()
+				Expect(config.Spec.PlacementConfiguration).To(Equal(currentPlacementConfiguration), "Should succeed deploying PlacementConfiguration on non pre-existing config")
+			})
+
+			It("should be able to update Placement-Configurations", func() {
```
Comment (rhrazdil) [inline test/e2e/workflow/deployment_test.go:159]: and here
```suggestion
			It("should be able to update PlacementConfiguration", func() {
```

File: test/e2e/workflow/deployment_test.go [hunk 4]
```diff
@@ -130,6 +130,57 @@ var _ = Describe("NetworkAddonsConfig", func() {
...
+					Infra: &cnao.Placement{ NodeSelector: map[string]string{
+						"kubernetes.io/hostname": "node01"},
+					},
+				}
+
+				configSpec := cnao.NetworkAddonsConfigSpec{
+					PlacementConfiguration: updatedPlacementConfiguration,
+				}
+				components := []Component{}
+				By("Re-deploying Placement-Configuration with different values")
```
Comment (rhrazdil) [inline test/e2e/workflow/deployment_test.go:173]: and here

File: test/e2e/workflow/deployment_test.go [hunk 5]
```diff
@@ -130,6 +130,57 @@ var _ = Describe("NetworkAddonsConfig", func() {
...
+				testConfigUpdate(configSpec, components)
+
+				By("Checking that deployed components remain up and running")
+				CheckConfigCondition(ConditionAvailable, ConditionTrue, CheckImmediately, time.Minute)
+
+				By("Checking Placement-Configuration was updated")
```
Comment (rhrazdil) [inline test/e2e/workflow/deployment_test.go:179]: and here

**Review by rhrazdil — CHANGES_REQUESTED** [review]

File: test/e2e/workflow/deployment_test.go (hunk 2 above)
Comment (RamLavi) [reply test/e2e/workflow/deployment_test.go]: Done

File: test/e2e/workflow/deployment_test.go (hunk 3 above)
Comment (RamLavi) [reply test/e2e/workflow/deployment_test.go]: Done

File: test/e2e/workflow/deployment_test.go (hunk 4 above)
Comment (RamLavi) [reply test/e2e/workflow/deployment_test.go]: Done

File: test/e2e/workflow/deployment_test.go (hunk 5 above)
Comment (RamLavi) [reply test/e2e/workflow/deployment_test.go]: Done

**Comment by RamLavi** [comment]
@rhrazdil fix [here](https://github.com/kubevirt/cluster-network-addons-operator/compare/766760ae4f20b7f1440fff7f89ef619c66b316fc..30ba32aa3d49caee22e93f7d159616eeb6cdbd55)

**Comment by rhrazdil** [comment]
@RamLavi Thanks!
/lgtm

**Review by phoracek — APPROVED** [review]
/lgtm
/approve

**Pull request merged**